File: CS/WindowsApplication1/CustomEditor/RepositoryItemMyButtonEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1 {
    [System.ComponentModel.DesignerCategory("")]
	[UserRepositoryItem("Register")]
	public class RepositoryItemMyButtonEdit : RepositoryItemButtonEdit { 
		static RepositoryItemMyButtonEdit() { 
			Register(); 
		}
		public RepositoryItemMyButtonEdit() 
        {
            TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.HideTextEditor;
        }
		
		internal const string EditorName = "MyButtonEdit";

        public override void CreateDefaultButton()
        {
            base.CreateDefaultButton();
            Buttons[0].Kind = DevExpress.XtraEditors.Controls.ButtonPredefines.Glyph;
            Buttons[0].Caption = "Test";
        }

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyButtonEdit),
                typeof(RepositoryItemMyButtonEdit), typeof(MyViewInfo),
                new ButtonEditPainter(), true, null));
        }
		public override string EditorTypeName { 
			get { return EditorName; } 
		}

	}
 
}

File: VB/WindowsApplication1/CustomEditor/RepositoryItemMyButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory(""), UserRepositoryItem("Register")> _
	Public Class RepositoryItemMyButtonEdit
		Inherits RepositoryItemButtonEdit
		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
			TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.HideTextEditor
		End Sub

		Friend Const EditorName As String = "MyButtonEdit"

		Public Overrides Sub CreateDefaultButton()
			MyBase.CreateDefaultButton()
			Buttons(0).Kind = DevExpress.XtraEditors.Controls.ButtonPredefines.Glyph
			Buttons(0).Caption = "Test"
		End Sub

		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyButtonEdit), GetType(RepositoryItemMyButtonEdit), GetType(MyViewInfo), New ButtonEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

	End Class

End Namespace